andretkachenko/nexus-bot
# Map and delete

# Description

- Add linked text channel autodeletion once voice channel is deleted
- Add map command

Fixes #119 #120 

## Type of change


# How Has This Been Tested?

1. Create voice channel `1`
2. Join `1`
3. Delete `1`
4. Ensure `1-text` was deleted

1. Create voice channel `1`
2. Create voice channel `2`
3. Join `1`
4. Join `2`
5. Map `1` and `2-text` via `map` command with override === 0
6. Observe `Voice Channel already has mapped Text Channel` error at the text channel where commands were sent

1. Create voice channel `1`
2. Create voice channel `2`
3. Join `1`
5. Map `2` and `1-text` via `map` command with override === 0
6. Observe `Text Channel is already mapped to a Voice Channel` error at the text channel where commands were sent

1. Create voice channel `1`
2. Create voice channel `2`
3. Join `1`
4. Join `2`
5. Map `1` and `2-text` via `map` command with override === 1
6. Observe `The Voice Channel already had a mapped Text Channel. Changed to map with the Text Channel #{id}` notification at the text channel where commands were sent
7. Observe `The Text Channel was already mapped to a Voice Channel. Deleted mapping with the Voice Channel #{id}` notification at the text channel where commands were sent
8. Observe `Mapped the Text Channel with the Voice Channel #{id}` notification at the text channel where commands were sent

1. type invalid command input: `sudo map 0 qwe qwe`
2. Observe `Error processing command - specified channel ID does not belong to a voice channel` error
3. Observe `invalid Voice Channel ID` error

1. Create voice channel `1`
2. Try to map `1` with invalid Text Channel ID
3. Observe `invalid Text Channel ID` error

**Test Configuration**:
* Firmware version: 0.0.307
* Hardware: Win 10
* Toolchain: VSCode, Discord, Google Chrome

# Checklist:
